tools/building.py

```diff
@@ -191,20 +191,22 @@ def lld_flags_for_executable(external_symbols):
   c_exports = [e for e in settings.EXPORTED_FUNCTIONS if is_c_symbol(e)]
   # Strip the leading underscores
   c_exports = [demangle_c_symbol_name(e) for e in c_exports]
-  c_exports += settings.EXPORT_IF_DEFINED
   # Filter out symbols external/JS symbols
   c_exports = [e for e in c_exports if e not in external_symbols]
+  c_exports += settings.REQUIRED_EXPORTS
   if settings.MAIN_MODULE:
     c_exports += side_module_external_deps(external_symbols)
   for export in c_exports:
-    cmd.append('--export-if-defined=' + export)
-
-  for export in settings.REQUIRED_EXPORTS:
-    if settings.ERROR_ON_UNDEFINED_SYMBOLS:
+    # We give special treatment to the `main` function since its
+    # always allowed to be undefined.
```

Comment:
Is main always allowed to be undefined, even if it is in EXPORTED_FUNCTIONS?

Reply:
I was being lazy here... refactored to avoid this special case.